src/main/java/edu/umich/its/cpm/Utils.java

```diff
@@ -672,6 +672,10 @@ public static String modifyFileNameOnType(String type, String fileName) {
 				fileName = fileName + Utils.HTML_FILE_EXTENSION;
 				return fileName;
 		}
+
+		if(fileExtension.equals("sav")){
```

Comment:
The reason for this check needs to be documented.

Consider pulling the code that deals with folder / file naming into a separate static class.  Having it in a separate class will make the processing much easier to find, explain, understand, and test this complicated code.